pkg/fanal/secret/builtin-rules.go

```diff
@@ -783,4 +784,13 @@ var builtinRules = []Rule{
 		SecretGroupName: "secret",
 		Keywords:        []string{"typeform"},
 	},
+	{
+		ID:              "azure-data-factory-registration-key",
+		Category:        CategoryAzure,
+		Title:           "Azure Data Factory Self-Hosted Integration Runtime registration key",
+		Severity:        "HIGH",
+		Regex:           MustCompile(`IR@[0-9a-zA-Z-]{36}@(.*?)@`),
```

Comment:
I checked this Regex and test secret.
Regex doesn't match full secret line - https://regex101.com/r/fPj6zd/1
Is this expected?